### PR TITLE
Use built-in OpenID Connect authentication

### PR DESCRIPTION
**Pull request by mderriey**
Hi there 👋

## TL;DR; breakdown by commits

1. use OpenID Connect authentication
1. leverage MVC capabilities
1. fix the sample when not running in IIS Express

### 1. Use OpenID Connect authentication
 
I noticed this sample implements an OpenID Connect flow manually.
While I appreciate this is a sample to show how to embed Power BI reports, I think rolling your own authentication is never a good idea, and I'm afraid people might use this implementation in their projects.

The main goal of this PR is to take advantage of the great authentication system in ASP.NET Core and leverage the OpenID Connect provider to delegate authentication to Azure AD.

There are small hacks &mdash; that I'll highlight as comments in this PR &mdash; because the redirect URL configured in the Azure AD application is set to the root of the application, `http://localhost:21638`. The recommendation is to use dedicated endpoints as redirect URLs because by default, the OIDC provider assumes all requests to the redirect URL will contain some sort of OpenID Connect message, which is not the case here as it also serves the page that embeds the Power BI report.
I'm not sure whether you have control over the app registration in Azure, but using something more common like `http://localhost:21638/signin-oidc` would greatly reduce the complexity of this sample; you can leave the existing redirect URL as-is to not break the experience for people who have already cloned or forked this repository.

### 2. Leverage MVC capabilities

Other small changes leverage the capabilities of ASP.NET Core MVC to redirect to a page or get the error message in the error page.

### 3. Fix the sample when not running in IIS Express

The `launchsettings.json` file describes which URLs are used when the project is run both in IIS Express and directly in the console through the Kestrel web server.
The application is hosted over a different port, `21639`, when running in Kestrel, which means when Azure AD tried to redirect the user to `http://localhost:21638`, there's no application running on that port.

### CODE REVIEW

File: .Net Core Sample/NetCore-Sample/Startup.cs [hunk 1]
```diff
@@ -21,7 +22,79 @@ public Startup(IConfiguration configuration)
...
+                    // See https://github.com/aspnet/AspNetCore/pull/10060
+                    options.Events.OnMessageReceived = context =>
+                    {
+                        if (context.Properties.Items.Count == 0)
+                        {
+                            context.SkipHandler();
+                        }
+
+                        return Task.CompletedTask;
+                    };
```
Comment (mderriey) [inline .Net Core Sample/NetCore-Sample/Startup.cs]: This is the "hack" section I'm mentioning in the PR description; using a dedicated endpoint for the redirect URL would make those go away.

**Comment by ali-hamud** [comment]
We are publishing new samples with most requests features and authentication/authorization flows. we are closing this PR.
Thank you.
